### ISSUE TITLE
Getting transformed images to be uploaded using a custom method

### ISSUE THREAD

**Issue by coluccini** [pqina/filepond-plugin-image-transform#10]
Hi! 
First of all: thanks for this amazing library ❤️

I'm using filepond (_the React component._) on a custom form and I want to upload the images when saving the whole for information. I'm saving files on component's state and when submitting the data I'm uploading each file and the save the data with the uploaded file urls on the server.
How do I get the transformed files on client size before uploading them to the server if I'm not using the `server` method?

This is a simplify version of my component:

```javascript
import React, { Component } from 'react';
import { FilePond, registerPlugin } from 'react-filepond';
import FilePondPluginFileValidateType from 'filepond-plugin-file-validate-type';
import FilePondPluginImageResize from 'filepond-plugin-image-resize';
import FilePondPluginImagePreview from 'filepond-plugin-image-preview';
import FilePondPluginImageTransform from 'filepond-plugin-image-transform';
import 'filepond/dist/filepond.min.css';
import 'filepond-plugin-image-preview/dist/filepond-plugin-image-preview.css';

registerPlugin(FilePondPluginFileValidateType,
  FilePondPluginImagePreview,
  FilePondPluginImageResize,
  FilePondPluginImageTransform,
);

class Form extends Component {
  constructor(props, context) {
    super(props, context);

    this.handleSubmit = this.handleSubmit.bind(this);
    this.onDropPhotos = this.onDropPhotos.bind(this);

    this.state = {
      title: '',
      photos: [],
    };
  }

  onDropPhotos() {
    const files = this.photosFilePond.getFiles();

    this.setState({
      photos: files.map(f => f.file),
    });
  }

  handleValueChange(field, value) {
    this.setState({ [field]: value });
  }

  async handleSubmit() {
    const {
      title, photos,
    } = this.state;

    const uploadedPhotos = [];
    if (photos.length) {
      await Promise.all(
        photos.map(async (file) => {
          const snapshot = await firebase.storage().ref().put(file);
          const downloadURL = await snapshot.ref.getDownloadURL();
          uploadedPhotos.push(downloadURL);
        }),
      );
    }

    const payload = {
      title,
      photos: uploadedPhotos,
    };
    // Save data on server
  }


  render() {
    const { title } = this.state;

    return (
      <div className="Form">
        <Input placeholder="title..." value={title} onChange={e => this.handleValueChange('title', e.target.value)} />
        <FilePond
          ref={ref => this.photosFilePond = ref} // eslint-disable-line no-return-assign
          allowMultiple
          allowImageResize
          allowImageTransform
          imageTransformOutputMimeType="image/jpeg"
          imageResizeUpscale={false}
          imageResizeTargetWidth={900}
          acceptedFileTypes={['image/png', 'image/jpeg']}
          imagePreviewHeight="100"
          onaddfile={this.onDropPhotos}
          onremovefile={this.onDropPhotos}
        />
        <Button onClick={this.handleSubmit}>Post</Button>
      </div>
    );
  }
}

export default Form;
```
Thanks in advance!

**Comment by rikschennink** [comment]
I think you can use the `onpreparefile` method to access the modified file data

**Comment by coluccini** [comment]
Thanks @rikschennink, but it seems that the method it's never triggered.
I've added this to the component `onpreparefile={(file, output) => console.log(file, output)}` and never gets printed. It could be that is not triggered until `processFiles()` is called?

I've also tried triggering `processFiles()` but looks like it doesn't work without `server` being set, cause it throws the following error `Unhandled Rejection (TypeError): Cannot read property 'url' of null`.
